pkg/splunk/enterprise/clustermaster_test.go

```diff
@@ -471,8 +683,116 @@ func TestAppFrameworkApplyClusterMasterShouldNotFail(t *testing.T) {
 		t.Errorf(err.Error())
 	}
 
-	_, err = ApplyClusterMaster(client, &cr)
+	splclient.RegisterS3Client("aws")
+
+	Etags := []string{"cc707187b036405f095a8ebb43a782c1"}
+	Keys := []string{"admin_app.tgz"}
+	Sizes := []int64{10}
+	StorageClass := "STANDARD"
+	randomTime := time.Date(2021, time.May, 1, 23, 23, 0, 0, time.UTC)
+
+	mockAwsHandler := spltest.MockAWSS3Handler{}
+
+	mockAwsObjects := []spltest.MockAWSS3Client{
+		{
+			Objects: []*spltest.MockAWSS3Object{
+				{
+					Etag:         &Etags[0],
+					Key:          &Keys[0],
+					LastModified: &randomTime,
+					Size:         &Sizes[0],
+					StorageClass: &StorageClass,
+				},
+			},
+		},
+	}
+
```

Comment:
`createRemoteObjectList` returns the type `[]*splclient.RemoteObject`, whereas the above code requires `[]*spltest.MockAWSS3Object` and the conversion from slice of one struct to slice of another is not straightforward. I might have to implement another method to go over each field and copy it to another struct. So I can take it up as a task as I mentioned in previous comment.

Reply:
As per our discussion, we can take it up as a separate task since this requires non-trivial changes and right now we are blocked because of circular dependency between splclient and spltest modules.